miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppPermission.kt

```diff
@@ -0,0 +1,22 @@
+package com.rakuten.tech.mobile.miniapp.js
+
+/** Type of miniapp permission. **/
+enum class MiniAppPermissionType(val type: String) {
+    UNKNOWN("unknown"),
+    LOCATION("location");
+
+    internal companion object {
+
+        internal fun getValue(type: String) = values().find { it.type == type } ?: UNKNOWN
+    }
+}
+
+internal enum class MiniAppPermissionResult(val type: String) {
+    ALLOWED("Allowed"),
+    DENIED("Denied");
```

Comment:
This should have a third type as well, depicting the state of `DO_NOT_ASK_AGAIN`. What do you think?

Reply:
There is a state of `DO_NOT_ASK_AGAIN` and from android 10 there is also a new state. However, the result from `PackageManager` only has two values: PERMISSION_GRANTED & PERMISSION_DENIED.

The result from both android & ios should only return two basic results.

Reply:
Thanks for sharing @minh-rakuten. I will check through the updates. 
It should have three results, accordingly, the Mini App could propose going to the Settings screen if required.

Platform differences are acceptable.
@ts-leojoseph-r @Climbatize is it handled the same way in iOS?

Reply:
@ts-nitish-a-jain  In iOS we are returning **Allowed**, **Denied**, **Not Determined** (When user taps on Not now)

Reply:
@minh-rakuten @ts-leojoseph-r @corycaywood 
I think the Result should model the superset of all the result cases. What do you guys think?
Android 11 has one-time permission and so does iOS.

Reply:
What android & ios are doing is:
- post `success` callback if permission is granted.
- post `error` callback if permission is denied. (whatever the reason is).

Android & iOS have different text results for different cases. I don't think the miniapp wants to handle all the result text cases. Miniapp just cares about whether the permission is granted or not.

Reply:
But mini-app would need to know if they need to ask for permission again or not for subsequent use.

Reply:
I believe miniapp should always ask for permission because user anytime can disable / enable it on phone Setting.

Reply:
Yeah, I think @minh-rakuten is right. We can require that the Mini App always request the permission before use of features. We discussed this during the architecture discussions and product side seemed to agree.

Reply:
@minh-rakuten you are right about that. 
From the pov of the code, it should speak the same language as it will be recommended in the docs; to be indicative that they must ask for permission before using the feature. Need to update the script code too for that.

This tells me that we should also expose a hasPermissionAccess method?

Reply:
@ts-nitish-a-jain Yeah, we need to update the document to mention that miniapp should always request permissions before getting data.

What does `hasPermissionAccess` do on mobile sdk side?

Reply:
I think that `hasPermissionAccess` will add a lot of complexity. We won't know what device permissions the Host App is using and the user could revoke the permission at any time, so we wouldn't be able to know anything about this on the SDK side. It would have to be another method that the Host App needs to integrate. Maybe we can leave it for a possible future enhancement if we can find a good way of doing it?

Reply:
Ok. Future is fine too.